packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -296,46 +461,59 @@ class _ZoomEnterTransition extends StatelessWidget {
     // the interrupted reversal of the forward transition to smoothly fade
     // the scrim away. This prevents a disjointed removal of the scrim.
     if (!reverse && animation.status != AnimationStatus.completed) {
-      opacity = _scrimOpacityTween.evaluate(animation)!;
+      scrimOpacity = _ZoomEnterTransitionState._scrimOpacityTween.evaluate(animation)!;
     }
 
-    final Animation<double> fadeTransition = reverse
-      ? kAlwaysCompleteAnimation
-      : _fadeInTransition.animate(animation);
+    if (scrimOpacity > 0.0) {
+      context.canvas.drawRect(
+        Offset.zero & Size(logicalWidth, logicalHeight),
+        Paint()..color = Colors.black.withOpacity(scrimOpacity),
+      );
+    }
+    _drawImageScaledAndCentered(context, image, scale.value, fade.value, pixelRatio);
+  }
 
-    final Animation<double> scaleTransition = (reverse
-      ? _scaleDownTransition
-      : _scaleUpTransition
-    ).animate(animation);
+  @override
+  void dispose() {
+    animation.removeListener(notifyListeners);
+    scale.removeListener(notifyListeners);
+    fade.removeListener(notifyListeners);
+    super.dispose();
+  }
 
-    return AnimatedBuilder(
-      animation: animation,
-      builder: (BuildContext context, Widget? child) {
-        return ColoredBox(
-          color: Colors.black.withOpacity(opacity),
-          child: child,
-        );
-      },
-      child: FadeTransition(
-        opacity: fadeTransition,
-        child: ScaleTransition(scale: scaleTransition, child: child),
-      ),
-    );
+  @override
+  bool shouldRepaint(covariant _ZoomEnterTransitionDelegate oldDelegate) {
+    return oldDelegate.reverse != reverse
+      || oldDelegate.animation != animation
+      || oldDelegate.scale != scale
+      || oldDelegate.fade != fade;
   }
 }
 
-class _ZoomExitTransition extends StatelessWidget {
+class _ZoomExitTransition extends StatefulWidget {
   const _ZoomExitTransition({
     required this.animation,
     this.reverse = false,
+    required this.preferRasterization,
     this.child,
   }) : assert(animation != null),
        assert(reverse != null);
 
   final Animation<double> animation;
+  final bool preferRasterization;
   final bool reverse;
   final Widget? child;
 
+  @override
+  State<_ZoomExitTransition> createState() => _ZoomExitTransitionState();
+}
+
+class _ZoomExitTransitionState extends State<_ZoomExitTransition> with _ZoomTransitionBase {
+  late _ZoomExitTransitionDelegate delegate;
+
+  @override
+  bool get allowRasterization => !kIsWeb && widget.preferRasterization;
```

Comment:
This is overly aggressive right?

I think you or yegor filed an issue somewhere for this, can we link it here with a TODO?

Reply:
Done